### ISSUE TITLE
solveset : expression having nested sqrt. 

### ISSUE THREAD

**Issue by Shekharrajak** [sympy/sympy#11184]
```
In [1]: solveset(20*sqrt(y**2 + (sqrt(-(y - 10)*(y + 10)) + 10)**2) - 60, y,S.Reals)
Out[1]: 
⎧-3⋅√391   3⋅√391⎫
⎨────────, ──────⎬
⎩   20       20  ⎭

In [2]: solve(20*sqrt(y**2 + (sqrt(-(y - 10)*(y + 10)) + 10)**2) - 60, y)
Out[2]: []
```

Ans is EmptySet.

**Comment by lePerdu** [comment]
As an intermediate step to solving (for `solveset()`), this comes up:
    `sqrt(-y**2 + 100) =  -191/20`.
The function `_invert_real()` keeps going with
    `-y**2 + 100 = 36481/400`,
ignoring that an even root cannot evaluate to a negative.

**Comment by oscarbenjamin** [comment]
The first example now gives an error:
```
In [5]: solveset(20*sqrt(y**2 + (sqrt(-(y - 10)*(y + 10)) + 10)**2) - 60, y,S.Reals)                                                                          
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-1770313d1461> in <module>
----> 1 solveset(20*sqrt(y**2 + (sqrt(-(y - 10)*(y + 10)) + 10)**2) - 60, y,S.Reals)

~/current/sympy/sympy/sympy/solvers/solveset.py in solveset(f, symbol, domain)
   1694             try:
   1695                 r = Dummy('r', **assumptions)
-> 1696                 return solveset(f.xreplace({symbol: r}), r, domain
   1697                     ).xreplace({r: symbol})
   1698             except InconsistentAssumptions:

~/current/sympy/sympy/sympy/solvers/solveset.py in solveset(f, symbol, domain)
   1712     f = piecewise_fold(f)
   1713 
-> 1714     return _solveset(f, symbol, domain, _check=True)
   1715 
   1716 

~/current/sympy/sympy/sympy/solvers/solveset.py in _solveset(f, symbol, domain, _check)
    929         return result
    930     else:
--> 931         lhs, rhs_s = inverter(f, 0, symbol)
    932         if lhs == symbol:
    933             # do some very minimal simplification since

~/current/sympy/sympy/sympy/solvers/solveset.py in <lambda>(f, rhs, symbol)
    882     # assign the solvers to use
    883     solver = lambda f, x, domain=domain: _solveset(f, x, domain)
--> 884     inverter = lambda f, rhs, symbol: _invert(f, rhs, symbol, domain)
    885 
    886     result = EmptySet()

~/current/sympy/sympy/sympy/solvers/solveset.py in _invert(f_x, y, x, domain)
    154 
    155     if domain.is_subset(S.Reals):
--> 156         x1, s = _invert_real(f_x, FiniteSet(y), x)
    157     else:
    158         x1, s = _invert_complex(f_x, FiniteSet(y), x)

~/current/sympy/sympy/sympy/solvers/solveset.py in _invert_real(f, g_ys, symbol)
    200         g, h = f.as_independent(symbol)
    201         if g is not S.Zero:
--> 202             return _invert_real(h, imageset(Lambda(n, n - g), g_ys), symbol)
    203 
    204     if f.is_Mul:

~/current/sympy/sympy/sympy/solvers/solveset.py in _invert_real(f, g_ys, symbol)
    219                 numer, denom = expo.as_numer_denom()
    220                 if denom % 2 == 0:
--> 221                     base_positive = solveset(base >= 0, symbol, S.Reals)
    222                     res = imageset(Lambda(n, real_root(n, expo)
    223                         ), g_ys.intersect(

~/current/sympy/sympy/sympy/solvers/solveset.py in solveset(f, symbol, domain)
   1712     f = piecewise_fold(f)
   1713 
-> 1714     return _solveset(f, symbol, domain, _check=True)
   1715 
   1716 

~/current/sympy/sympy/sympy/solvers/solveset.py in _solveset(f, symbol, domain, _check)
    924         try:
    925             result = solve_univariate_inequality(
--> 926             f, symbol, domain=domain, relational=False)
    927         except NotImplementedError:
    928             result = ConditionSet(symbol, f, domain)

~/current/sympy/sympy/sympy/solvers/inequalities.py in solve_univariate_inequality(expr, gen, relational, domain, continuous)
    516                 # this might raise ValueError on its own
    517                 # or it might give None...
--> 518                 solns = solvify(e, gen, domain)
    519                 if solns is None:
    520                     # in which case we raise ValueError

~/current/sympy/sympy/sympy/solvers/solveset.py in solvify(f, symbol, domain)
   1797 
   1798         else:
-> 1799             solution = solution_set.intersect(domain)
   1800             if isinstance(solution, FiniteSet):
   1801                 result += solution

~/current/sympy/sympy/sympy/sets/sets.py in intersect(self, other)
    106 
    107         """
--> 108         return Intersection(self, other)
    109 
    110     def intersection(self, other):

~/current/sympy/sympy/sympy/sets/sets.py in __new__(cls, *args, **kwargs)
   1217         # Reduce sets using known rules
   1218         if evaluate:
-> 1219             return simplify_intersection(args)
   1220 
   1221         return Basic.__new__(cls, *args)

~/current/sympy/sympy/sympy/sets/sets.py in simplify_intersection(args)
   1918             if len(other_sets) > 0:
   1919                 other = Intersection(other_sets)
-> 1920                 return Union(Intersection(arg, other) for arg in s.args)
   1921             else:
   1922                 return Union(*[arg for arg in s.args])

~/current/sympy/sympy/sympy/sets/sets.py in __new__(cls, *args, **kwargs)
   1025                 return sum(map(flatten, arg), [])
   1026             raise TypeError("Input must be Sets or iterables of Sets")
-> 1027         args = flatten(args)
   1028 
   1029         # Union of no sets is EmptySet

~/current/sympy/sympy/sympy/sets/sets.py in flatten(arg)
   1023                     return [arg]
   1024             if iterable(arg):  # and not isinstance(arg, Set) (implicit)
-> 1025                 return sum(map(flatten, arg), [])
   1026             raise TypeError("Input must be Sets or iterables of Sets")
   1027         args = flatten(args)

~/current/sympy/sympy/sympy/sets/sets.py in flatten(arg)
   1023                     return [arg]
   1024             if iterable(arg):  # and not isinstance(arg, Set) (implicit)
-> 1025                 return sum(map(flatten, arg), [])
   1026             raise TypeError("Input must be Sets or iterables of Sets")
   1027         args = flatten(args)

~/current/sympy/sympy/sympy/sets/sets.py in <genexpr>(.0)
   1918             if len(other_sets) > 0:
   1919                 other = Intersection(other_sets)
-> 1920                 return Union(Intersection(arg, other) for arg in s.args)
   1921             else:
   1922                 return Union(*[arg for arg in s.args])

~/current/sympy/sympy/sympy/sets/sets.py in __new__(cls, *args, **kwargs)
   1217         # Reduce sets using known rules
   1218         if evaluate:
-> 1219             return simplify_intersection(args)
   1220 
   1221         return Basic.__new__(cls, *args)

~/current/sympy/sympy/sympy/sets/sets.py in simplify_intersection(args)
   1942             new_args = False
   1943             for t in args - set((s,)):
-> 1944                 new_set = intersection_sets(s, t)
   1945                 # This returns None if s does not know how to intersect
   1946                 # with t. Returns the newly intersected set otherwise

~/current/sympy/sympy/sympy/multipledispatch/dispatcher.py in __call__(self, *args, **kwargs)
    196             self._cache[types] = func
    197         try:
--> 198             return func(*args, **kwargs)
    199 
    200         except MDNotImplementedError:

~/current/sympy/sympy/sympy/sets/handlers/intersection.py in intersection_sets(self, other)
    272         return imageset(Lambda(n_, re),
    273                         self.base_set.intersect(
--> 274                             solveset_real(im, n_)))
    275 
    276     elif isinstance(other, Interval):

~/current/sympy/sympy/sympy/solvers/solveset.py in solveset_real(f, symbol)
   1716 
   1717 def solveset_real(f, symbol):
-> 1718     return solveset(f, symbol, S.Reals)
   1719 
   1720 

~/current/sympy/sympy/sympy/solvers/solveset.py in solveset(f, symbol, domain)
   1712     f = piecewise_fold(f)
   1713 
-> 1714     return _solveset(f, symbol, domain, _check=True)
   1715 
   1716 

~/current/sympy/sympy/sympy/solvers/solveset.py in _solveset(f, symbol, domain, _check)
    952                         result += _solve_radical(equation,
    953                                                  symbol,
--> 954                                                  solver)
    955                     elif equation.has(Abs):
    956                         result += _solve_abs(f, symbol, domain)

~/current/sympy/sympy/sympy/solvers/solveset.py in _solve_radical(f, symbol, solveset_solver)
    749         f_set = []  # solutions for FiniteSet
    750         c_set = []  # solutions for ConditionSet
--> 751         for s in result:
    752             if checksol(f, symbol, s):
    753                 f_set.append(s)

~/current/sympy/sympy/sympy/sets/sets.py in __iter__(self)
   1161             return roundrobin(*(iter(arg) for arg in self.args))
   1162         else:
-> 1163             raise TypeError("Not all constituent sets are iterable")
   1164 
   1165 class Intersection(Set):

TypeError: Not all constituent sets are iterable
```

**Comment by jksuom** [comment]
Empty set is probably correct for `S.Reals` as the argument of `sqrt` is bigger that `10**2` for all real `y` and therefore `20*sqrt(... + 10**2) - 60 > 20*10 - 60 > 0`.

I now get this complicated expression for the empty set.
```
>>> solveset(20*sqrt(y**2 + (sqrt(-(y - 10)*(y + 10)) + 10)**2) - 60, y,S.Reals)
Union(Intersection(Interval(-10, 10), ImageSet(Lambda(x, sqrt(100 - x**2)), EmptySet())), Intersection(Interval(-10, 10), ImageSet(Lambda(x, -sqrt(100 - x**2)), EmptySet())))
```

**Comment by eagleoflqj** [comment]
Fixed in 651595d137c2c4469d97be2458f469dfbdb2c9b9 from https://github.com/sympy/sympy/pull/21276
